iot2050-firmware-update: Refactor and add map3 firmware update

huaqianli opened the pull request:
iot2050-firmware-update: Refactor and add map3 firmware update

jan-kiszka commented on recipes-app/iot2050-firmware-update/files/iot2050-firmware-update.tmpl:
NACK, I'm using that all the time to update the freshly build binaries directly.

jan-kiszka commented:
FWIW: copy `flash-pgX.bin` / `iot2050-pgX-image-boot.bin` to target, run `iot2050-firmware-update -fn that.bin` on the target is a developer use case for the flashing tool that we should preserve. It already asks you often enough to confirm what you do.

jan-kiszka commented:
No commit message for first patch, and a major diff for that one - not really reviewable. What are the goals of this rewrite? Can those be achieved in smaller steps?

huaqianli commented:
> > No commit message for first patch, and a major diff for that one - not really reviewable. What are the goals of this rewrite? Can those be achieved in smaller steps?
> 
> At the beginning, we planed to add the map3 firmware into this tarball and expected to upgrade firmware as a whole, Which is why I removed the "--force" option.
> 
> When I added the map3 firmware, I found that we had to refactor the code for better maintainability and scalability. So I refactored it. There are no new features to be introduced, just refactoring.

Yesterday, we had some new findings, after a meeting we decided to upgrade map3 firmware with another way and keep the refactoring code without map3 part.

jan-kiszka commented:
Whenever you touch code, explain the reason. And the more you touch it, the longer the reasoning should be. "For better maintainability" is a bit too short here.
